python-libraries/data-platform-catalogue/data_platform_catalogue/entities.py

```diff
@@ -102,11 +108,14 @@ class TableMetadata:
     description: str
     column_details: list
     retention_period_in_days: int | None
+    domain: str | None = None
```

Comment:
Do you think we still need the Data Product upsert method now? We wrote it with the assumption we were building a new data platform ingestion mechanism, and I think some of the assumptions are no longer valid. Datahub lets you manage data products as yaml out of the box without writing any python code, so I feel like this is not adding anything at the moment.